packages/shell-api/src/mongo.ts

```diff
@@ -451,22 +453,34 @@ export default class Mongo extends ShellApiClass {
 
   @topologies([Topologies.ReplSet])
   startSession(options: Document = {}): Session {
-    const driverOptions = {};
-    if (options === undefined) {
-      return new Session(this, driverOptions, this._serviceProvider.startSession(driverOptions));
+    const allTransactionOptions = [
+      'readConcern', 'writeConcern', 'readPreference', 'maxCommitTimeMS'
+    ] as const;
+    // These typechecks might look weird, but will tell us if we are missing
+    // support for a newly introduced driver option when it is being added
+    // to the driver API.
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const __typecheck1: (typeof allTransactionOptions)[number] = '' as Exclude<keyof TransactionOptions, keyof CommandOperationOptions>;
```

Comment:
Don't have a strong preference really, but it took me awhile to grasp what's exactly is going on there, at the same time I used something like this before (which provides the same result and also doesn't require eslint-disable):

```ts
    function assertAllOptionsUsed(options: Exclude<keyof TransactionOptions, keyof CommandOperationOptions>) {
          // These typechecks might look weird, but will tell us if we are missing
          // support for a newly introduced driver option when it is being added
          // to the driver API.
    }
    assertAllOptionsUsed(allTransactionOptions)
```

Reply:
Well, your suggestion is basically to use a function call instead of an assignment, right? That’s easy enough to do, updated the code here :) Still needs eslint comments, though, because the argument is unused.